docs/source/quickstart.mdx

```diff
@@ -116,7 +147,24 @@ Use the [`~Dataset.set_format`] function to set the dataset format to `torch` an
 ```
 </pt>
 <tf>
-Use the [`~Dataset.to_tf_dataset`] function to set the dataset format to be compatible with TensorFlow. You'll also need to import a [data collator](https://huggingface.co/docs/transformers/main_classes/data_collator#transformers.DataCollatorWithPadding) from 🤗 Transformers to combine the varying sequence lengths into a single batch of equal lengths:
+
+Use the [`~TFPreTrainedModel.prepare_tf_dataset`] method from 🤗 Transformers to prepare the dataset to be compatible with
+TensorFlow, and ready to train/fine-tune a model, as it wraps a HuggingFace [`~datasets.Dataset`] as a `tf.data.Dataset`
+with collation and batching, so one can pass it directly to Keras methods like `fit()` without further modification. 
+
+```py
+>>> import tensorflow as tf
+
+>>> tf_dataset = model.prepare_tf_dataset(
+...     dataset,
+...     batch_size=4,
+...     shuffle=True,
+... )
+```
+
+Additionally, you can also use the [`~Dataset.prepare_tf_dataset`] function to set the dataset format to 
+be compatible with TensorFlow; and you'll also need to import a [data collator](https://huggingface.co/docs/transformers/main_classes/data_collator#transformers.DataCollatorWithPadding) 
+from 🤗 Transformers to combine the varying sequence lengths into a single batch of equal lengths:
```

Comment:
`to_tf_dataset` is explained in the `Use with Tensorflow` doc. Also, we plan to rewrite this method completely to make it faster, easier to use, support nested data, etc.. So, in the current state, it's OK not to "promote" it here:

````suggestion
Use the [`~TFPreTrainedModel.prepare_tf_dataset`] method from 🤗 Transformers to prepare the dataset to be compatible with TensorFlow. This method also handles collation and batching, making the dataset ready for training: 

```py
>>> import tensorflow as tf

>>> tf_dataset = model.prepare_tf_dataset(
...     dataset,
...     batch_size=4,
...     shuffle=True,
... )
```
````

PS: For ``[`~TFPreTrainedModel.prepare_tf_dataset`]`` to be resolved, `transformers` and `tensorflow` must be added as doc dependencies to `setup.py. You can also add `torch` to the list for the DataLoader reference (and remove the hardcoded link)

Reply:
> So, in the current state, it's OK not to "promote" it here

Yeah maybe we can remove it entirely? I think for the Quickstart it's not as important to present alternatives and we can just stick to `prepare_tf_dataset`.